src/NuGet.Core/NuGet.ProjectModel/ProjectLockFile/PackagesLockFileUtilities.cs

```diff
@@ -66,13 +67,20 @@ public static string GetNuGetLockFilePath(string baseDirectory, string projectNa
         /// </summary>
         /// <param name="dgSpec">The <see cref="DependencyGraphSpec"/> for the new project defintion.</param>
         /// <param name="nuGetLockFile">The current <see cref="PackagesLockFile"/>.</param>
-        /// <returns>True if the lock file is valid false otherwise. </returns>
-        public static bool IsLockFileStillValid(DependencyGraphSpec dgSpec, PackagesLockFile nuGetLockFile)
+        /// <returns>Returns LockFileValidityWithInvalidReasons object with IsValid set to true if the lock file is valid false otherwise.
+        /// The second return type is a localized message that indicates in further detail the reason for the inconsistency.</returns>
+        public static LockFileValidityWithInvalidReasons IsLockFileStillValid(DependencyGraphSpec dgSpec, PackagesLockFile nuGetLockFile)
         {
+            List<string> invalidReasons = new List<string>();
+
             // Current tools know how to read only previous formats including the current
             if (PackagesLockFileFormat.PackagesLockFileVersion < nuGetLockFile.Version)
             {
-                return false;
+                invalidReasons.Add(string.Format(
```

Comment:
I know I said didn't love the approach, but your point about you need to take an action anyways convinced me otherwise. 

I'd hate for you keep chnaging these things back and forth.
What do you think?

Reply:
AFAIK, the chances of user hitting this case are extremely low. I expect them to hit other cases such as `new tfm or changes in the dependency` invalidating the lock file. We can leave this for now if we are good with other changes in this PR.